tests/test_web_application.py

```diff
@@ -91,6 +91,36 @@ def test_non_default_router():
     assert router is app.router
 
 
+class MockAbstractRouter(AbstractRouter):
+
+    @abstractmethod
+    def add_resource(self, oath, name=None):
+        pass
+
+
+def test_add_route(loop):
+    router = mock.Mock(spec=MockAbstractRouter)
+    app = web.Application(router=router)
+
+    @app.route('/index.html')
```

Comment:
in my own impplementation of such decorator, I add reference to decorated method and all decorator's kwargs and args  to some global array. And also I have a function that adds this saved information into application's routes.

In short (pseudocode):
```python
global_array = []
def handler(*args, **kwargs):
    def decorator(function):
        global_array.append((function, args, kwargs))
        return function
    return decorator

@handler(....)
async def myhandler(request):
    ....

class MyApplication(web.Application):
    def add_saved_routes(self):
        for (method, args, kwargs):
            self.add_route(....)

def main():
    app = MyApplication()
    app.add_saved_routes()
    ...
```

This allows me to use decorators BEFORE creating application (during imports). This is especially important since I have many handlers which are placed in many files.

So, could you provide complete example showing usage of your decorators for the case when there are many files with handlers ?